packages/dapp/src/components/Listings.tsx

```diff
@@ -39,19 +40,29 @@ class Listings extends React.Component<{}, ListingsState> {
     return (
       <StyledDiv>
         whaddup listings: {this.state.applications.toString()}
+        <br/>
+        {this.state.error && this.state.error}
       </StyledDiv>
     );
   }
 
-  private initListings = () => {
-
+  private initListings = async () => {
     const civil = new Civil();
-    const tcr = civil.tcrSingletonTrusted();
-    const instance = this;
-    const subscription = tcr.listingsInApplicationStage().subscribe((listing) => {
-      instance.setState({applications: this.state.applications.push(listing)});
-    });
-    this.setState({applicationSubscription : subscription});
+    let tcr;
+    try {
+      tcr = civil.tcrSingletonTrusted();
+    } catch (ex) {
+      console.log("failed to get tcr.");
```

Comment:
For the future - might want to compare what error you got before showing a nice message, and display the error if not recognized